lib/FusionInventory/Agent/Task/Inventory/Linux/Networks/DockerMacvlan.pm

```diff
@@ -0,0 +1,90 @@
+package FusionInventory::Agent::Task::Inventory::Linux::Networks::DockerMacvlan;
+
+use strict;
+use warnings;
+
+use parent 'FusionInventory::Agent::Task::Inventory::Module';
+
+use JSON::PP;
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Network;
+
+sub isEnabled {
+    return canRun('docker');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my @networks = _getMacvlanNetworks(logger => $logger);
+
+    foreach my $network (@networks) {
+        my @interfaces = _getInterfaces(logger => $logger, networkId => $network);
+        foreach my $interface (@interfaces) {
+            $inventory->addEntry(
+                section => 'NETWORKS',
+                entry   => $interface
+            );
+        }
+    }
+}
+
+sub _getMacvlanNetworks {
+    my (%params) = (
+        command => 'docker network ls --filter driver=macvlan -q',
+        @_
+    );
+    my $handle = getFileHandle(%params);
+    return unless $handle;
+
+    my @networks;
+
+    while (my $line = <$handle>) {
+        chomp($line);
+        push @networks, $line;
+    }
+
+    close $handle;
+    return @networks;
+}
+
+sub _getInterfaces {
+    my (%params) = @_;
+
+    $params{command} = "docker network inspect $params{networkId}";
```

Comment:
You should better here do the same way than in _getMacvlanNetworks method, l.36-39

Reply:
```perl
sub _getInterfaces {
    my (%params) = (
        command => 'docker network inspect',
        @_
    );

    $params{command} .= ' ' . $params{networkId};
```

Is this what you want?

Reply:
Not exactly... the purpose of the `( command => '...', @_)` syntax is to permit to test the method as `command` parameter could be overridden by a `command => 'cat command_output_dump_file'`.
But finally, as you are testing using the `file` member in `dockermacvlan.t`, the syntax is okay until the `file` member is always used first in getFileHandle() API.
If you want you can better replace the code by:
```
    my (%params) = @_;

    my $lines = getAllLines(
        command => "docker network inspect $params{networkId}",
        %params
    ) or return;
```
But do as you wish, I'll accept the current code.